darwin/datatypes.py

```diff
@@ -19,6 +21,38 @@
 ErrorHandler = Callable[[int, str], None]
 
 
+@dataclass(frozen=True)
+class StringDataSource:
+    """
+    Represents a source from a ``StringData`` class.
+    """
+
+    #: The id of the source.
```

Comment:
I personally don't like these comment lines, they don't add anything to what the code is already saying

Reply:
This is part of the documentation and avoids an error. These are not comment lines, these are documentation lines. 
The reason behind switching to `#:` is explained here: https://www.notion.so/v7labs/Python-Code-Style-351a7831c2074e78bf4e21ec13777542

(basically, a bug in Sphinx forces us to document dataclasses this way)